pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -68,6 +70,17 @@ func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
 }
 
 func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry, raw json.RawMessage, apiVersion string, resourceID string) api.ValidationResult {
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusNotFound) {
```

Comment:
It took me a hot second to understand why we perform this specific callout of `http.StatusNotFound` here but not with `isCreate` below.

For future reviewers: in my reading, our `Get` implementation always returns `nil` when `err` is not `nil` (including NotFound), so there's a case here where we get `nil` + `http.StatusNotFound`, which translates into `isCreate := true`, so the logic is sound with how `Get` happens to be implemented today - you just have to read all of `Get` to know it.

However, even though the current logic is sound as `Get` is written, I think `isCreate := cosmosdb.IsErrorStatusCode(err, http.StatusNotFound)` is a clearer and less brittle way to show that this specific error case is the only time where isCreate should be `true`:

```
doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
isCreate := cosmosdb.IsErrorStatusCode(err, http.StatusNotFound)
if err != nill && !isCreate {
    // return err
}

// rest of the changes
```

Reply:
Made the suggested changes